source/kraken/tests/data_manager_with_type_data_test.cpp

```diff
@@ -0,0 +1,121 @@
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+Stay tuned using
+twitter @navitia
+IRC #navitia on freenode
+https://groups.google.com/d/forum/navitia
+www.navitia.io
+*/
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE data_manager_test
+#include <boost/test/unit_test.hpp>
+
+#include "kraken/data_manager.h"
+#include "type/data.h"
+
+static const std::string fake_data_file = "fake_data.nav.lz4";
+static const std::string fake_disruption_path = "fake_disruption_path";
+
+// We create a empty data with lz4 format in current directory.
+#define CREATE_FAKE_DATA(fake_file_name) \
+    navitia::type::Data data(0);         \
+    data.save(fake_data_file);           \
+
+// Absolute path
+std::string complete_path(const std::string file_name){
+    char buf[256];
+    if (getcwd(buf, sizeof(buf))) {
+        return std::string(buf) + "/" + file_name;
+    } else {
+        std::perror("getcwd");
+        return "";
+    }
+}
+
+// test sequence :
+// 1. load good data
+// 2. reload good data
+// 3. reload wrong data
+BOOST_AUTO_TEST_CASE(load_success) {
+
+    CREATE_FAKE_DATA(fake_data_file)
+
+    DataManager<navitia::type::Data> data_manager;
+    BOOST_CHECK(data_manager.get_data());
+    auto first_data = data_manager.get_data();
+
+    // Same pointer
+    BOOST_CHECK_EQUAL(first_data, data_manager.get_data());
+
+    // real data : Loading successed
+    BOOST_CHECK(data_manager.load(complete_path(fake_data_file)));
+
+    // Load success, so the internal shared pointer change.
+    auto second_data = data_manager.get_data();
+    BOOST_CHECK_NE(first_data, second_data);
+
+    // Same pointer
+    BOOST_CHECK_EQUAL(second_data, data_manager.get_data());
+
+    // When we reload a new data.
+    // real data : Loading successed
+    BOOST_CHECK(data_manager.load(complete_path(fake_data_file)));
+
+    // Load success, so the internal shared pointer change.
+    auto third_data = data_manager.get_data();
+    BOOST_CHECK_NE(second_data, third_data);
+
+    // Same pointer
+    BOOST_CHECK_EQUAL(third_data, data_manager.get_data());
+
+    // When we reload a new data.
+    // Fake data : Loading failed
+    BOOST_CHECK(!data_manager.load("wrong path"));
+
+    // Load failed, so the internal shared pointer no change.
+    // Data has not changed.
+    auto fourth_data = data_manager.get_data();
+    BOOST_CHECK_EQUAL(third_data, fourth_data);
+}
+
+BOOST_AUTO_TEST_CASE(load_failed) {
```

Comment:
Isn't that teste redundant with the end of the previous one?
Also, you did not test for all the branches of the load:
* only nav, and nav fail are done
* missing the parts where we smartly reload (or not)